File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin() , points.end() ,[](const auto& a , const auto& b){return a[1] < b[1];});
        int right_limit = points[0][1];
        int arr = 1;
        for(const auto& i: points){
            if(i[0] > right_limit){
                ++arr;
                right_limit = i[1];
            }
        }
        return arr;
    }
};

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key = lambda x:x[1])
        rl = points[0][1]
        arr = 1
        for i in points:
            if i[0] > rl:
                arr += 1
                rl = i[1]
        return arr